github-cli.md

```diff
@@ -0,0 +1,110 @@
+
+
+## **Github**
+<img src="https://memegenerator.net/img/instances/41740049/regular-peopleadd-me-on-facebook-me-fork-me-on-github.jpg" width="300px" height="300px"/>
+
+### Introduction
+**What: ** Git is a version control system which allows multiple people to work on a project at the same time (kind of like Google Docs).
+
+**Why: ** Imagine cooking with your friends: you all have to keep track of what the others are adding to the dish while making sure that what you add doesn't ruin it. Things get more complicated as more and more people join. The same is true of technical projects.
+
+**How: ** [Github!](https://github.com/)
+
+### Commands
+#### `git init <repository name`
+This will create a new git repository on your **local** machine
+
+#### `git clone <link to Github repository>`
+This allows you to create a copy of the project on your local machine. When you make changes, they'll only show up on your copy, we'll talk about how to **push** your changes to the remote repository in another command
+
+#### `git checkout [-b <new branch name> | <existing branch>]`
+<img src="https://wac-cdn.atlassian.com/dam/jcr:86eba9ec-9391-45ea-800a-948cec1f2ed7/Branch-2.png?cdnVersion=lf" width="400px" height="300px"/>
+
+Branching is useful when you want to add a feature to the project without affecting the code of others working on it. It allows you to work on your feature without worrying about what others are working on *for now*
+
+#### `git add [<file/directory> | .]`
+Adding a file means that you want to move a file from the working directory to the staging directory. These files will be added to the next **commit** which will be explained in the next command. **Note: ** `git add .` indicates that you want to add all the files you just updated to the staging directory.
+
+#### `git commit -m "<commit message>"`
+This command is used to save the changes you made in your local repository. The commit message is often a sentence or two explaining what you changed and why.
+
+#### `git status`
+This command will show you what branch you're on, tracked and untracked files, and which files are staged for a commit.
+
+#### `git push origin <branch name>`
+This command is used to update the remote repository (on Github) with the changes you just made on your local repository.
+
+#### `git pull origin <branch name>`
+This allows you to update your local copy of the repository with any changes others updated the remote repository with. **Do this every now and then so that your branch is not completely outdated**
+
+#### `git rebase <branch>`
+Sometimes more than one person will edit the same file and a **merge conflict** will occur. Use this command to choose which and whose changes you want to include in your file.
+
+#### `git merge <branch name>`
+This command will allow you to combine two branches together. **BE CAREFUL!!! You typically want at least one other person to check over your code to make sure that your changes don't break the project**
```

Comment:
merge master is the stressful one for resolving conflicts. You can mention merge conflicts here. You can say, if you're on master, merging in another branch can be dangerous because people expect master to have working code. --> this is typically why PRs exist.

Worth mentioning the whole formality may be dropped at a hackathon, but it's worth keeping these processes in mind and it's still useful for a team to coordiante even on the timescale of 36 hours.